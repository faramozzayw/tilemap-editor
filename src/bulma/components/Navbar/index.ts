export { NavbarItem } from "./NavbarItem";
export { NavbarDropdown } from "./NavbarDropdown";
export { NavbarDivider } from "./NavbarDivider";

export { NavbarStart } from "./NavbarStart";
export { NavbarEnd } from "./NavbarEnd";
export { NavbarLink } from "./NavbarLink";
