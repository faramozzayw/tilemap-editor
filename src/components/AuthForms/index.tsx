export { LogIn } from "./Login";
export { LogOut } from "./LogOut";
export { SignUp } from "./SignUp";
