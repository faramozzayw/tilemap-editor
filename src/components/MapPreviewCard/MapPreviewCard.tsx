import React, { CSSProperties, useState, useEffect } from "react";
import { useHistory, Link } from "react-router-dom";
import { Transition } from "react-transition-group";

import { Title, Card, CardContent, Image } from "@faramo.zayw/reabulma";

import "./index.css";
import { MapConfig } from "./MapConfig";
import { IAuth } from "../../types/auth";
import { UserLink, MapName, Can, CoolBox, MarkdownRemark } from "../../common";
import { Map } from "../../types/graphql";

import { Key } from "./MapConfig";

import fakeImage from "./fake.jpg";

export interface MapPreviewCardProps extends Map, IAuth {}

const duration = 200;

const defaultStyle: CSSProperties = {
	transition: `opacity ${duration}ms ease-in-out`,
	opacity: 0,
};

const transitionStyles: any = {
	entering: { opacity: 1 },
	entered: { opacity: 1 },
	exiting: { opacity: 0 },
	exited: { opacity: 0 },
};

/*
<Can role={isAuth ? "user" : ""} perform="like:map"></Can>
*/

/*
title={<MapName name={name} to={`/maps/${id}`} />}
*/

export const MapPreviewCard: React.FC<MapPreviewCardProps> = ({
	author,
	name,
	id,
	isAuth,
	description,
	size,
	...props
}) => {
	const history = useHistory();
	const [inProp, setInProp] = useState(false);
	const [liked, setLiked] = useState(false);

	useEffect(() => {
		setTimeout(() => setInProp(true), duration);
	}, [setInProp]);

	if (!author) {
		return null;
	}

	return (
		<Transition in={inProp} timeout={duration}>
			{(state) => (
				<article
					className="MapPreviewCard"
					style={{
						...defaultStyle,
						...transitionStyles[state],
					}}
				>
					<CoolBox>
						<Link to={`/maps/${id}`}>
							<Image src={fakeImage} />
						</Link>
						<div className="mapContent">
							<div className="summary">
								<p>
									<Key>Map size:</Key> {size.row} x {size.column}
								</p>
								<p>
									<Key>Create:</Key>{" "}
									<time dateTime="2016-1-1">
										{new Date(props.createdAt).toLocaleDateString()}
									</time>
								</p>
								<p>
									<Key>Last edit:</Key>{" "}
									<time dateTime="2016-1-1">
										{props.updatedAt
											? new Date(props.updatedAt).toLocaleDateString()
											: "not edited yet"}
									</time>
								</p>
								<p>
									<Key>Auhtor:</Key> <UserLink username={author.username} />
								</p>
							</div>
							<MarkdownRemark markdown={description} />
						</div>
						<div className="activitySummary">
							<span className="icon">
								<i className="fas fa-database"></i>
								<span>142mb</span>
							</span>
							<span className="icon">
								<i className="fas fa-heart"></i>
								<span>150k</span>
							</span>
							<span className="icon">
								<i className="fas fa-download"></i>
								<span>150k</span>
							</span>
						</div>
					</CoolBox>
				</article>
			)}
		</Transition>
	);
};
