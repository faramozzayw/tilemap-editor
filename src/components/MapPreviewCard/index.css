.MapPreviewCard .card {
	border-radius: 0;
}

.MapPreviewCard:hover {
	box-shadow: 3px 4px 6px 1px #272727;
}

.MapPreviewCard a,
.MapPreviewCard button {
	color: #fff;
}

.MapPreviewCard .config {
	padding: 1rem;
	border-radius: 0.25rem;
	background-color: #3e3e3e;
}

.MapPreviewCard .card-footer {
	border: none;
}

.MapPreviewCard .card-footer-item {
	border: none;
}

.MapPreviewCard .card-footer .button {
	border-radius: 0;
}

.MapPreviewCard .like-button {
	background: transparent;
	border: none;
	font-size: 2rem;
}

.MapPreviewCard .like-button:hover {
	cursor: pointer;
}

.MapPreviewCard .like-button:focus,
.MapPreviewCard .like-button:focus-within {
	outline: 2px dashed indianred;
}

.MapPreviewCard .image-wrapper {
	background: linear-gradient(180deg, #88c767, #545050, #4a4a4a);
	padding: 0.25rem;
	border-radius: 0.25rem 0.25rem 0 0;
}

.MapPreviewCard .image {
	z-index: 10;
	transition: transform 0.35s ease-in;
}

.MapPreviewCard .image img {
	box-shadow: 18px -3px 0px 13px rgb(0 0 0 / 55%);
}

.MapPreviewCard .image:hover,
.MapPreviewCard .image:focus {
	cursor: pointer;
	transform: scale(1.5) rotate(1turn);
}

.masonry-grid {
	display: flex;
	margin-left: -30px;
	width: auto;
}

.masonry-grid_column {
	padding-left: 30px;
	background-clip: padding-box;
}

.masonry-grid_column > article {
	background: transparent;
	margin-bottom: 30px;
}
