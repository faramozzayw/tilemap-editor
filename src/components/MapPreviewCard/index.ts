export { MapFeed } from "./MapFeed";
export { MapPreviewCard } from "./MapPreviewCard";
