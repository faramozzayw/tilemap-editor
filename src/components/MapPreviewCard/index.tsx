import React from "react";
import { Link, useHistory } from "react-router-dom";

import { Button } from "./../../bulma";
import { MapConfig } from "./../../types";
import { deleteMap } from "./../../store/mapsStore";

import "./index.css";

export interface MapPreviewCardProps extends MapConfig {}

export const MapPreviewCard: React.FC<MapPreviewCardProps> = ({
	author,
	name,
	description,
	size,
	create_data,
	last_edit,
	id,
}) => {
	const history = useHistory();

	const editHandler = () => history.push(`/editor/${id}`);
	const deleteHandler = () => deleteMap(id);

	return (
		<div className="MapPreviewCard card has-background-grey-dark has-text-primary-light">
			<div className="card-content">
				<div className="media">
					<div className="media-left">
						<figure className="image is-48x48">
							<img
								src="https://bulma.io/images/placeholders/96x96.png"
								alt="Placeholder image"
							/>
						</figure>
					</div>
					<div className="media-content">
						<p className="title is-4">{name}</p>
						<p className="subtitle is-6">@{author}</p>
					</div>
				</div>

				<div className="content">
					{description}
					{description && <br />}
					Map size: {size.row} x {size.column}
					<br />
					Create:{" "}
					<time dateTime="2016-1-1">{create_data.toLocaleDateString()}</time>
					<br />
					Last edit:{" "}
					<time dateTime="2016-1-1">
						{last_edit?.toLocaleString() ?? "not edited yet"}
					</time>
				</div>
			</div>
			<footer className="card-footer">
				<Button
					className="card-footer-item"
					isColor="success"
					isOutlined
					onClick={editHandler}
				>
					<Link to={`/editor/${id}`}>Edit</Link>
				</Button>
				<Button
					className="card-footer-item"
					isColor="danger"
					onClick={deleteHandler}
				>
					Delete
				</Button>
			</footer>
		</div>
	);
};
