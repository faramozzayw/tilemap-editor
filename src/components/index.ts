export { EditorNavbar } from "./EditorNavbar";
export { EditorTabs } from "./EditorTabs";
export { EditorCanvas } from "./EditorCanvas";

export { MainNavbar } from "./MainNavbar";
