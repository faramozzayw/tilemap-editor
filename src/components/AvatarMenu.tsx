import React from "react";

import { NavbarItem } from "../bulma";
import { LogOut } from "./AuthForms";
import { NavbarDropdown, NavbarDivider } from "../bulma";

export interface AvatarMenuProps {
	image?: string;
	username: string;
}

const AvatarMenu: React.FC<AvatarMenuProps> = ({ username, image }) => {
	return (
		<NavbarItem isHoverable hasDropdown>
			<NavbarItem>
				{image && (
					<figure className="image is-32x32">
						<img src={image} className="is-rounded" alt="User picture" />
					</figure>
				)}
				<span>{username}</span>
			</NavbarItem>
			<NavbarDropdown className="is-right has-background-dark">
				<NavbarItem>Profile</NavbarItem>
				<NavbarDivider />
				<NavbarItem>
					<LogOut />
				</NavbarItem>
			</NavbarDropdown>
		</NavbarItem>
	);
};

export default AvatarMenu;
