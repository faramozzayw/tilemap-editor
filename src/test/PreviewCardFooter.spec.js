import React from "react";
import { render } from "@testing-library/react";

import { PreviewCardFooter } from "./../components/MapPreviewCard";
import { BrowserRouter } from "react-router-dom";

const props = {
	id: "iu2h3u284u2s9-234242s2342",
	editHandler: () => {},
	forkHandler: () => {},
	viewHandler: () => {},
};

test("renders correctly", () => {
	const { baseElement } = render(
		<BrowserRouter>
			<PreviewCardFooter {...props} />
		</BrowserRouter>,
	);

	expect(baseElement).toMatchSnapshot();
});
