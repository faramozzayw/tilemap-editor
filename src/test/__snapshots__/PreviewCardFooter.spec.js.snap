// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<body>
  <div>
    <footer
      class="card-footer"
    >
      <button
        class="button is-success card-footer-item"
        type="button"
      >
        <span>
          <i
            class="far fa-eye"
          />
           View
        </span>
      </button>
      <button
        class="button is-warning card-footer-item"
        type="button"
      >
        <span>
          <i
            class="fas fa-code-branch"
          />
           Fork
        </span>
      </button>
    </footer>
  </div>
</body>
`;
